deno/lib/__tests__/string.test.ts

```diff
@@ -166,9 +166,17 @@ test("url validations", () => {
   const url = z.string().url();
   url.parse("http://google.com");
   url.parse("https://google.com/asdf?asdf=ljk3lk4&asdf=234#asdf");
+  url.parse(
+    "https://anonymous:flabada@developer.mozilla.org/en-US/docs/Web/API/URL/password"
+  );
   expect(() => url.parse("asdf")).toThrow();
+  expect(() => url.parse("http:.......///broken.com")).toThrow();
+  expect(() => url.parse("c:")).toThrow();
+  expect(() => url.parse("WWW:WWW.COM")).toThrow();
   expect(() => url.parse("https:/")).toThrow();
+  expect(() => url.parse("https://asdf")).toThrow();
   expect(() => url.parse("asdfj@lkjsdf.com")).toThrow();
+  expect(() => url.parse("http://localhost")).toThrow();
```

Comment:
`http://localhost` is a valid URL.
It translates to `http://localhost:80/`
Same goes for `https://localhost`, that goes to `https://localhost:443`